### PR TITLE
Fix nested form validation

### PR DESCRIPTION
**Pull request by sliminas**
This pull request fixes issue https://github.com/trailblazer/reform-rails/issues/50, where form.validate returns true alltough there is a nested form with errors present.

### CODE REVIEW

File: test/form_test.rb [hunk 1]
```diff
@@ -12,4 +12,21 @@ class SongForm < Reform::Form
...
+  Song = Struct.new(:length)
+  class PopulatedAlbumForm < Reform::Form
+    property :hit, populate_if_empty: Song do
+      property :length
+      validates :length, numericality: { greater_than: 55 }
+    end
+  end
+  it do
+    form = PopulatedAlbumForm.new(Album.new)
+    form.validate({ :hit => { :length => "54" }})
```
Comment (seuros) [inline test/form_test.rb]: I think this line is duplicate

**Comment by apotonick** [comment]
Really important: we also have to keep the same case in `reform` to make sure the Dry code doesn't run into the same problem.

**Comment by aiomaster** [comment]
@apotonick Would be cool to merge this!

**Comment by apotonick** [comment]
@fran-worley Shouldn't we rather rethink the Result object?

**Comment by aganov** [comment]
Any progress on this? This issue is very worrying one... Its breaking the main functionality of reform - Validating... :(

**Comment by apotonick** [comment]
Has anyone tested this?

**Comment by aganov** [comment]
@apotonick I've tested @sliminas's branch, seems to working fine with `reform v2.3.0.rc1`

**Comment by aganov** [comment]
@apotonick Although it seems like you might have abandoned this repo, can we at least have this merged and get some `v0.2.0.rc2` release please...

**Comment by apotonick** [comment]
Sure, it's not abandoned! We're simply not enough people to deliver fast enough in this growing eco-system!

**Comment by fernandes** [comment]
thank you @sliminas !

**Comment by sliminas** [comment]
@fernandes Nice to seeing this merged. Glad to help =)

**Comment by fernandes** [comment]
@sliminas released v0.2.0.rc2 😉

**Pull request merged**